packages/wrangler/src/versions/deploy.ts

```diff
@@ -558,11 +558,12 @@ async function promptPercentages(
 async function maybePatchSettings(
 	accountId: string,
 	workerName: string,
-	config: Pick<Config, "logpush" | "tail_consumers">
+	config: Pick<Config, "logpush" | "tail_consumers" | "observability">
 ) {
 	const maybeUndefinedSettings = {
 		logpush: config.logpush,
 		tail_consumers: config.tail_consumers,
+		observability: config.observability, // TODO reconcile with how regular deploy handles empty state
```

Comment:
What's the desired end state here when it's empty?

Reply:
I'm not sure. When @RamIdeas added versions-deploy, any undefined settings are not synchronized. When @zebp added observability to regular deploy, it defaults to false when undefined.

It would be unfortunate to need to always patch settings to false for undefined observabililty, but that is the simplest way to reconcile them.

Reply:
If the default is false then how about we remove the default (make it undefined, i.e. falsey) so this just works™️

Reply:
@RamIdeas I'm not sure I follow. The default does seem to be undefined/falsey, but the difference between deploy and versions-deploy is that deploy always patches the full NonVersionedScriptSettings whereas versions-deploy only does so if at least one field is defined.

So deploy with undefined observability will disable it, whereas versions-deploy will leave it unchanged. Should we always patch during versions-deploy? Or conditionally patch during deploy? Or leave them slightly different?

Reply:
I don't think we should block on this. The behaviour here comes down to `wrangler deploy` doing something different for OBS compared to logpush/tail consumers, it will disable if it is not in the config. Meanwhile, logpush/tail will stay with their current values.

If we should do the explicit disable here or if deploy should not do that is a whole topic. The fact that OBS is already consistent isn't great, I'm happy with it at least being consistent here. I think there's a larger discussion that needs to happen (cc: @tanushree-sharma @nevikashah ) on what the behaviour should actually be in deploy & here.

I think part of this whole "user expectation" behaviour comes to the fact that Wrangler is already kinda inconsistent with these things. If you remove a binding from the config, it isn't inherited, it's removed. However, you need to explicitly disable workers.dev, it isn't disabled on not being there.

Imo, we need to figure out what the behaviour actually should be and be as consistent across the board as possible.

Reply:
Yep no need to block on this. Sounds like there's a backend inconsistency re the observability field? Either way, I agree we need to decide the desired behaviour cc @tanushree-sharma @nevikashah

Reply:
The backend API is consistent, the inconsistency is whether it's called or not by wrangler. `wrangler deploy` always calls it, `wrangler versions deploy` only calls if if one of the 3 observability settings are defined in wrangler.toml